From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony <tonymckahan@gmail.com>
Date: Sun, 15 Apr 2018 14:40:58 -0400
Subject: [ARCHEOLOGY] Rockchip next sync master and development

> X-Git-Archeology: > recovered message: > includes kernel panic fix on veth interfaces and audio device patch to enable Tinkerboard audio
> X-Git-Archeology: - Revision b1a21a44cde3d976b9b8ebce4f265e5aa79b2d8a: https://github.com/armbian/build/commit/b1a21a44cde3d976b9b8ebce4f265e5aa79b2d8a
> X-Git-Archeology:   Date: Sun, 15 Apr 2018 14:40:58 -0400
> X-Git-Archeology:   From: Tony <tonymckahan@gmail.com>
> X-Git-Archeology:   Subject: Rockchip next sync master and development
> X-Git-Archeology:
> X-Git-Archeology: - Revision b6f1503dfdb215429cc6dee6c2b94ca9ddc96fbc: https://github.com/armbian/build/commit/b6f1503dfdb215429cc6dee6c2b94ca9ddc96fbc
> X-Git-Archeology:   Date: Mon, 26 Nov 2018 01:38:25 -0500
> X-Git-Archeology:   From: Thomas McKahan <tonymckahan@gmail.com>
> X-Git-Archeology:   Subject: [ tinkerboard ] dev kernel add audio support
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision f08dcd48677d2a34f349bf571c979cd422bffcc3: https://github.com/armbian/build/commit/f08dcd48677d2a34f349bf571c979cd422bffcc3
> X-Git-Archeology:   Date: Tue, 31 Oct 2023 08:13:23 +0100
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip,rk322x: bump edge kernel to 6.6 (#5875)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 562d96128ba6a511a8a06c0f4d29946ab80b8969: https://github.com/armbian/build/commit/562d96128ba6a511a8a06c0f4d29946ab80b8969
> X-Git-Archeology:   Date: Tue, 26 Dec 2023 16:45:30 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: consolidate rk322x and rockchip 32 bit families
> X-Git-Archeology:
> X-Git-Archeology: - Revision 47d2e8287e34fed3e47f37ab076d0f34ed0ac399: https://github.com/armbian/build/commit/47d2e8287e34fed3e47f37ab076d0f34ed0ac399
> X-Git-Archeology:   Date: Mon, 25 Mar 2024 19:38:38 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: bump edge kernel to 6.8
> X-Git-Archeology:
> X-Git-Archeology: - Revision 724573bf7a21e61b0b626f835031a4c3206bb8ba: https://github.com/armbian/build/commit/724573bf7a21e61b0b626f835031a4c3206bb8ba
> X-Git-Archeology:   Date: Wed, 05 Jun 2024 22:18:51 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip family edge kernel to 6.9
> X-Git-Archeology:
> X-Git-Archeology: - Revision 7da7bbf61cb776a054219e35926d391dad9a67a7: https://github.com/armbian/build/commit/7da7bbf61cb776a054219e35926d391dad9a67a7
> X-Git-Archeology:   Date: Mon, 22 Jul 2024 19:18:14 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: bump edge kernel to 6.10
> X-Git-Archeology:
> X-Git-Archeology: - Revision 94ec783de0dad381b3e2e71d646d8428af4d5051: https://github.com/armbian/build/commit/94ec783de0dad381b3e2e71d646d8428af4d5051
> X-Git-Archeology:   Date: Wed, 18 Sep 2024 14:03:19 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip 32 bit edge kernel to 6.11
> X-Git-Archeology:
> X-Git-Archeology: - Revision c90a0f7890bddc8e755847fc8227e15828950251: https://github.com/armbian/build/commit/c90a0f7890bddc8e755847fc8227e15828950251
> X-Git-Archeology:   Date: Sat, 30 Nov 2024 13:07:31 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: bump edge kernel to 6.12
> X-Git-Archeology:
---
 sound/usb/card.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/usb/card.c b/sound/usb/card.c
index 111111111111..222222222222 100644
--- a/sound/usb/card.c
+++ b/sound/usb/card.c
@@ -533,6 +533,14 @@ static void usb_audio_make_shortname(struct usb_device *dev,
 	}
 
 	strim(card->shortname);
+
+	/* Tinker Board ALC4040 CODEC	*/
+
+	if(USB_ID_VENDOR(chip->usb_id) == 0x0bda &&
+		USB_ID_PRODUCT(chip->usb_id) == 0x481a) {
+			strlcat(card->shortname, " OnBoard", sizeof(card->shortname));
+	}
+	
 }
 
 static void usb_audio_make_longname(struct usb_device *dev,
-- 
Armbian

